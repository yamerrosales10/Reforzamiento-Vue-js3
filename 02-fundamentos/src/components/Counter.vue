<template>
    <h2>Counter2</h2>
    <p>{{ numero }} <sup>2</sup> = {{ mult() }}</p>
    <p data-testid="counter">{{ numero }}</p>
    <button @click="increment">+</button>
    <button @click="decrement">-</button>
</template>

<script>
export default {
    name: "Counter",
    props: {
        // number: Number,
        number: {
            type: Number,
            required: true,
            default: 3, //default(){return{a:"s"}}

            validator(value) {
                return value > 100;
            },
        },
    },
    data() {
        return {
            numero: this.number,
        };
    },
    methods: {
        mult() {
            console.log("xd");
            return this.numero * this.numero;
        },
        increment() {
            this.numero++;
        },
        decrement() {
            this.numero--;
        },
    },
    computed: {
        mult2() {
            console.log("xdcomputed");
            return this.numero * this.numero;
        },
    },
    // setup: {
    //     mult() {
    //         return this.numero * this.numero;
    //     },
    // },
};
</script>

<style></style>
