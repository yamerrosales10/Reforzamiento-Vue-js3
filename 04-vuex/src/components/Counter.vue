<template>
    <h1>Counter - Vuex</h1>
    <h2>Direct Access {{ $store.state.counter.count }}</h2>
    <!-- <h2>Computed {{ countComputed }}</h2> -->
    <h1>mapState</h1>
    <h2>mapState:{{ count }}</h2>
    <div class="flex gap-2">
        <button class="bg-red-600 px-2 text-white" @click="increment">
            +1
        </button>
        <button class="bg-green-600 px-2 text-white" @click="incrementBy(5)">
            +5
        </button>
        <button
            :disabled="isLoading"
            class="bg-blue-600 px-2 text-white"
            @click="incrementRandomInt"
        >
            {{ isLoading ? "cargando" : "random" }}
        </button>
    </div>
    <!-- <h2>Direct getter: {{ $store.getters.getterquepusimos  }}</h2> -->
</template>
<script>
import { mapState, mapActions, mapGetters } from "vuex";
export default {
    // computed: mapState(["count"]),
    computed: {
        countComputed() {
            return this.$store.state.counter.count;
        },
        ...mapState("counter", ["count", "isLoading"]),
        ...mapGetters([""]),
        // ...mapState({count:state=>state.count)}) tambien asi se puede
        // ...mapState({count:"count"}) tambien asi se puede
    },
    methods: {
        increment() {
            this.$store.commit("counter/increment");
        },
        incrementBy(number) {
            this.$store.commit("counter/incrementBy", number);
        },
        // incrementRandomInt() {
        //     this.$store.dispatch("incrementRandomInt");
        // },
        // ...mapActions(["incrementRandomInt"]),
        ...mapActions("counter", ["incrementRandomInt"]),
    },
};
</script>
