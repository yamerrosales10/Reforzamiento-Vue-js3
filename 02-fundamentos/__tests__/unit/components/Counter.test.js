import Counter from "@/components/Counter.vue";
import { shallowMount } from "@vue/test-utils";
import { beforeEach, expect } from "vitest";
describe("Counter.vue ", () => {
    // it("should to snapshot match", () => {
    //     const wrapper = shallowMount(Counter);
    //     expect(wrapper.html()).toMatchSnapshot();
    // });
    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(Counter);
    });

    it("h2 Debe de tener el valor por defecto", () => {
        expect(wrapper.find("h2").exists()).toBeTruthy();
        const h = wrapper.find("h2").text();
        expect(h).toBe("Counter2");
    });

    it("El valor por defecto debe ser 100 en p", () => {
        const ptags = wrapper.find("[data-testid='counter'");
        // h[1].html  todo el html
        expect(ptags.text()).toBe("3");
    });

    it("debe de incremente en 1 el valor del contador", async () => {
        const incrementbtn = wrapper.find("button");
        await incrementbtn.trigger("click");
        const ptags = wrapper.find("[data-testid='counter'");
        // h[1].html  todo el html
        expect(ptags.text()).toBe("4");
    });
});
