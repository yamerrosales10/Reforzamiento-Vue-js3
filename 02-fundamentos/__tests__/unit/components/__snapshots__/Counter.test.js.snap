// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter.vue  > should to snapshot match 1`] = `
"<h2>Counter2</h2>
<p>3 <sup>2</sup> = 9</p>
<button>+</button>
<button>-</button>"
`;
